store/store.go

```diff
@@ -1359,7 +1359,22 @@ func (s *Store) Download(ctx context.Context, name string, targetPath string, do
 		url = downloadInfo.DownloadURL
 	}
 
-	err = download(ctx, name, downloadInfo.Sha3_384, url, user, s, w, resume, pbar)
+	if downloadInfo.Size == 0 || resume < downloadInfo.Size {
+		err = download(ctx, name, downloadInfo.Sha3_384, url, user, s, w, resume, pbar)
+	} else if downloadInfo.Sha3_384 != "" {
```

Comment:
what I mean, if you think we cannot error, it seems more convervative to behave as before in the case == "", not do a random thing as with this code